src/Model/Collection/FlattenedBlockSequence.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace eLife\ApiSdk\Model\Collection;
+
+use eLife\ApiSdk\CanBeCounted;
+use eLife\ApiSdk\Collection;
+use eLife\ApiSdk\Collection\ArraySequence;
+use eLife\ApiSdk\Collection\Sequence;
+use eLife\ApiSdk\ImmutableArrayAccess;
+use eLife\ApiSdk\Model\Block;
+use eLife\ApiSdk\Model\HasContent;
+use IteratorAggregate;
+use IteratorIterator;
+use Traversable;
+
+final class FlattenedBlockSequence implements IteratorAggregate, Sequence
+{
+    use CanBeCounted;
+    use ImmutableArrayAccess;
+
+    private $sequence;
+
+    public function __construct(Sequence $sequence)
+    {
+        $blockFinder = function (array $array, Block $block) use (&$blockFinder) {
```

Comment:
would make this a (probably private) method to let it being referenced as `[$this, "blockFinder"]` in recursion rather than resorting to `&`